tests/shap/test_shap.py

```diff
@@ -40,3 +84,63 @@ def test_log_matplotlib_figure():
 
     artifacts = set(yield_artifacts(run.info.run_id))
     assert artifacts == {"test.png", "dir/test.png"}
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("artifact_path", [None, "dir"])
+def test_log_explanation_with_regressor(regressor, artifact_path):
+    model = regressor.model
+    X = regressor.X
+
+    with mlflow.start_run() as run:
+        explanation_path = mlflow.shap.log_explanation(model.predict, X, artifact_path)
+
+    artifact_path_expected = "shap" if artifact_path is None else artifact_path
+    artifacts = set(yield_artifacts(run.info.run_id))
+
+    assert explanation_path == os.path.join(run.info.artifact_uri, artifact_path_expected)
+    assert artifacts == {
+        os.path.join(artifact_path_expected, "base_values.npy"),
+        os.path.join(artifact_path_expected, "shap_values.npy"),
+        os.path.join(artifact_path_expected, "summary_bar_plot.png"),
+    }
+
+    shap_values = np.load(os.path.join(explanation_path, "shap_values.npy"))
+    base_values = np.load(os.path.join(explanation_path, "base_values.npy"))
+
+    explainer = shap.KernelExplainer(model.predict, shap.kmeans(X, 100))
+    shap_values_expected = explainer.shap_values(X)
+    base_values_expected = np.array(explainer.expected_value)
+
+    np.testing.assert_array_equal(shap_values, shap_values_expected)
+    np.testing.assert_array_equal(base_values, base_values_expected)
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("artifact_path", [None, "dir"])
+def test_log_explanation_with_classifier(classifier, artifact_path):
+    model = classifier.model
+    X = classifier.X
+
+    with mlflow.start_run() as run:
+        explanation_path = mlflow.shap.log_explanation(model.predict_proba, X, artifact_path)
+
+    artifact_path_expected = "shap" if artifact_path is None else artifact_path
+    artifacts = set(yield_artifacts(run.info.run_id))
+
+    assert explanation_path == os.path.join(run.info.artifact_uri, artifact_path_expected)
+    assert artifacts == {
+        os.path.join(artifact_path_expected, "base_values.npy"),
+        os.path.join(artifact_path_expected, "shap_values.npy"),
+        os.path.join(artifact_path_expected, "summary_bar_plot.png"),
+    }
+
+    shap_values = np.load(os.path.join(explanation_path, "shap_values.npy"))
+    base_values = np.load(os.path.join(explanation_path, "base_values.npy"))
+
+    explainer = shap.KernelExplainer(model.predict_proba, shap.kmeans(X, 100))
+    shap_values_expected = explainer.shap_values(X)
+    base_values_expected = np.array(explainer.expected_value)
+
+    np.testing.assert_array_equal(shap_values, shap_values_expected)
+    np.testing.assert_array_equal(base_values, base_values_expected)
```

Comment:
I'm investigating whether or not `KernelExplainer` is really deterministic (I thought it isn't)